### ISSUE TITLE
DOC/TST: Clarify usage of r_squred

### ISSUE THREAD

**Issue by adriangb** [adriangb/scikeras#196]
`KerasRegressor.r_squared` is supposed to be used as a metric with Keras, but it is not tested as such and it is incorrectly used as a loss function in some places. This is very bad. 2 things are required to fix this:
1. Add a test that checks that it works as a metric.
2. Replace every usage of it as a loss function with mean squared error, including in notebooks.
